resources/static/js/account.js

```diff
@@ -86,72 +94,90 @@ function showRamp() {
 }
 
 async function checkWeb3Network() {
-  let required_chain_id = CHAIN_ID;
-  await connectWeb3();
-  web3.version.getNetwork(function (error, chain_id) {
-    if (error) {
-      console.error(error);
-    }
-
-    if (chain_id != required_chain_id) {
-      let current_chain_name = CHAIN_ID_MAPPING[chain_id];
-      let required_chain_name = CHAIN_ID_MAPPING[required_chain_id];
-      alert(
-        `Web3 Browser connected to ${current_chain_name}, please change to ${required_chain_name}.`
-      );
-    }
-  });
+  const requiredChainID = CHAIN_ID;
+  const currentChainID = parseInt(await provider.request({ method: 'eth_chainId' }), 16);
```

Comment:
Before there was console error logging. Shouldn't the await call be wrapped around a try/catch block?

Reply:
Good point! Thanks.